base/src/protocol_state_proof/opening_proof.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::types::{FieldElement, FiniteECPoint, FiniteECPointPairVec};
+use serde::{Deserialize, Serialize};
+use wire_type::WireType;
+
+use ark_ec::models::ModelParameters;
+use ark_ec::short_weierstrass_jacobian::GroupAffine;
+
+#[derive(Clone, Serialize, Deserialize, Default, PartialEq, Debug, WireType)]
+#[serde(from = "<Self as WireType>::WireType")]
+#[serde(into = "<Self as WireType>::WireType")]
+pub struct OpeningProof {
+    pub lr: FiniteECPointPairVec,
+    pub z_1: FieldElement,
+    pub z_2: FieldElement,
+    pub delta: FiniteECPoint,
+    pub sg: FiniteECPoint,
+}
+
+impl<P> From<OpeningProof> for commitment_dlog::commitment::OpeningProof<GroupAffine<P>>
+where
+    P: ark_ec::SWModelParameters,
+    <P as ModelParameters>::BaseField: From<ark_ff::BigInteger256>,
+    <P as ModelParameters>::ScalarField: From<ark_ff::BigInteger256>,
+{
+    fn from(t: OpeningProof) -> Self {
```

Comment:
Just to note, there seems to be no code coverage for all newly added from trait impls, we need to improve that if not applicable in this PR.

https://app.codecov.io/gh/ChainSafe/mina-rs/compare/81/diff#D4L28